mtp/mtp_fs.cpp

```diff
@@ -4,336 +4,320 @@
+        .access_caps = MTP_STORAGE_READ_WRITE,
+        .capacity    = 0,
+        .description = "Storage",
+        .volume_id   = "1234567890abcdef",
+    };
+
+    constexpr auto iobuf_size = 64U * 1024U; // 64kB
+
+    bool is_dot(const char *name)
     {
-        return static_cast<std::uint64_t>(MTP_SPARE_SPACE);
+        const auto name_length = strlen(name);
+        const bool ret =
+            ((name_length == 1) && (name[0] == '.')) || ((name_length == 2) && (memcmp(name, "..", 2) == 0));
+        log_debug("is_dot: '%s': %s", name, ret ? "true" : "false");
+        return ret;
     }
-    constexpr auto iobuf_size = 64U * 1024U;
-}
-
-
-typedef struct {
-    uint32_t read;
-    uint64_t capacity;
-    uint64_t freespace;
-} fs_data_t;
-
-static int is_dot(const char *name) {
-    bool rv = ((strlen(name) == 1) && (name[0] == '.')) ||
-        ((strlen(name) == 2) && (name[0] == '.') && (name[1] == '.'));
-    LOG("is_dot: %s: %s", name, rv ? "true" : "false");
-    return rv;
-}
 
-static uint32_t count_files(DIR * find_data)
-{
-    uint32_t count = 0;
-    rewinddir(find_data);
-    struct dirent *de;
-    while ((de=readdir(find_data)))
+    uint32_t count_files(DIR *find_data)
     {
-        if (is_dot(de->d_name))
-            continue;
-       count++;
+        uint32_t count = 0;
+        rewinddir(find_data);
+        struct dirent *de;
+        while ((de = readdir(find_data)) != nullptr) {
+            if (is_dot(de->d_name)) {
+                continue;
+            }
+            count++;
+        }
+        return count;
     }
-    return count;
-}
-
-static const char *abspath(const char *rootPath, const char *filename)
-{
-    static char abs[128];
-    abs[0] = '/';
-    strncpy(&abs[1], filename, 64);
-    snprintf(abs, 128, "%s/%s", rootPath, filename);
-    return abs;
-}
 
-static const mtp_storage_properties_t* get_disk_properties(void* arg)
-{
-    fs_data_t *data = (fs_data_t*)arg;
-
-    struct statvfs stvfs {};
-    statvfs( purefs::dir::getUserDiskPath().c_str(), &stvfs);
-
-    // TODO: stats are for entire storage. If MTP is intended to expose
-    // only one directory, these stats should be recalculated
-    const auto freeSpace = stvfs.f_bsize * stvfs.f_bavail;
-    data->freespace = (getSpareSpace() > freeSpace) ? 0 : freeSpace - getSpareSpace();
-    data->capacity =  stvfs.f_frsize * stvfs.f_blocks;
+    const mtp_storage_properties_t *get_disk_properties(void *arg)
+    {
+        const struct mtp_fs *fs = static_cast<struct mtp_fs *>(arg);
+        struct statvfs stvfs {};
+        statvfs(fs->root, &stvfs);
+
+        [[maybe_unused]] const auto freeSpace = stvfs.f_bavail * stvfs.f_bsize;
+        const auto capacity                   = stvfs.f_blocks * stvfs.f_bsize;
+        disk_properties.capacity              = capacity;
+
+        log_debug("Capacity: %u MB, free: %u MB",
+                  static_cast<unsigned>(capacity / 1024 / 1024),
```

Comment:
A minor thing, but may be confusing for someone: these are de facto MiBs, not MBs.